csrc/preseg_passes/propagate_shardings.cpp

```diff
@@ -19,6 +19,182 @@
 namespace nvfuser::preseg_passes {
 
 namespace {
+std::pair<std::unordered_set<IterDomain*>, std::unordered_set<IterDomain*>>
+getReshapedIds(
+    ViewOp* view_op,
+    const std::unordered_map<IterDomain*, IterDomain*>& c2p) {
+  std::unordered_set<IterDomain*> p_reshaped_ids; // Reshaped logical IDs
+  std::unordered_set<IterDomain*> c_reshaped_ids; // Reshaped root IDs
+
+  TensorView* consumer = view_op->out();
+  std::vector<IterDomain*> c_root_domain = consumer->getMaybeRootDomain();
+
+  for (auto id : consumer->getLogicalDomain()) {
+    if (id->isRFactorProduct() && id->definition() &&
+        !id->definition()->isA<Resize>()) {
+      auto root_ids = getInputsInTargetDomain(id, c_root_domain);
+      for (auto root_id : root_ids) {
+        c_reshaped_ids.insert(root_id);
+      }
+    }
+  }
+
+  for (auto id : c_reshaped_ids) {
+    if (auto p_id = c2p.find(id); p_id != c2p.end()) {
+      p_reshaped_ids.insert(p_id->second);
+    }
+  }
+  return std::make_pair(p_reshaped_ids, c_reshaped_ids);
+}
+
+bool isSplitDivisible(IterDomain* id, Split* split) {
```

Comment:
Split::isDivisible, e.g., https://github.com/NVIDIA/Fuser/blob/d0ecedd37a797d84b8082e2196e46e67accda5ef/csrc/device_lower/utils.cpp#L1484

Reply:
Got it -- it's a dry-run